lxd/container_lxc.go

```diff
@@ -3189,12 +3181,12 @@ func (c *containerLXC) OnStop(target string) error {
 	return nil
 }
 
-// cleanupNetworkDevices performs any needed network device cleanup steps when container is stopped.
-func (c *containerLXC) cleanupNetworkDevices(netns string) {
+// cleanupDevices performs any needed device cleanup steps when container is stopped.
+func (c *containerLXC) cleanupDevices(netns string) {
```

Comment:
@stgraber this function has been expanded to deal with cleaning up nic and proxy devices. I'm still running it from the OnStopNS hook for now, because nics need that for now (until liblxc detaches hotplugged devices), and this hook is run before the OnStop hook so the proxy is still stopped before the storage is stopped.